openapi.yaml

```diff
@@ -132,120 +580,313 @@ securityDefinitions:
+      type: &pizzaType
+        type: string
+        description: the type of the resource (pizza)
+        enum: [pizza]
+      id: &pizzaId
+        type: string
+        description: a unique ID of a pizza
+        example: abc123
+      links:
+        $ref: '#/definitions/SelfLink'
+      attributes:
+        $ref: '#/definitions/PizzaAttributes'
+      relationships: &pizzaRelationships
+        properties:
+          dough:
+            required: [data]
+            properties:
+              links:
+                properties:
+                  self:
+                    type: string
+                    description: a link to the relationship
+                    example: https://api.oregonstate.edu/v1/pizzas/1/relationships/dough
+                  related:
+                    type: string
+                    description: a link to the dough referenced
+                    example: https://api.oregonstate.edu/v1/pizzas/1/dough
+              data:
+                $ref: '#/definitions/DoughRelationshipResource'
+          ingredients:
+            required: [data]
+            properties:
+              links:
+                properties:
+                  self:
+                    type: string
+                    description: a link to the relationship
+                    example: https://api.oregonstate.edu/v1/pizzas/1/relationships/dough
+                  related:
+                    type: string
+                    description: a link to the ingredients for the dough referenced
+                    example: https://api.oregonstate.edu/v1/pizzas/1/ingredients
+              data:
+                type: array
+                description: a list of relationship resources with the ingredient type
+                items:
+                  $ref: '#/definitions/IngredientRelationshipResource'
+      included: &pizzaIncluded
+        type: array
+        items:
+          type: object
+          description: An ingredient resource or a dough resource
+  PizzaPostBody:
+    required: [data]
+    properties:
+      data:
+        required: [type, attributes]
+        properties:
+          type: *pizzaType
+          attributes:
+            $ref: '#/definitions/PizzaAttributes'
+          relationships: &pizzaUpdateRelationships
+            properties:
+              dough:
+                properties:
+                  data:
+                    $ref: '#/definitions/DoughRelationshipResource'
+              ingredients:
+                properties:
+                  data:
+                    type: array
+                    items:
+                      $ref: '#/definitions/IngredientRelationshipResource'
+  PizzaPatchBody:
+    required: [data]
+    properties:
+      data:
+        required: [id, type]
+        properties:
+          type: *pizzaType
+          id: *pizzaId
+          attributes:
+            $ref: '#/definitions/PizzaAttributes'
+          relationships: *pizzaUpdateRelationships
+  PizzaPostAttributes:
```

Comment:
Looks like this definition is defined but never used. Should this be removed?

Reply:
This should be used on line 659 instead of `"#/definitions/PizzaAttributes"`, in the definition of PizzaPostBody. I'll add it there